app/editor/src/features/admin/topic-score-rules/TopicScoreRuleList.tsx

```diff
@@ -0,0 +1,250 @@
+import { FormikForm } from 'components/formik';
+import { useLookupOptions } from 'hooks';
+import React from 'react';
+import { FaArrowDown, FaArrowUp, FaTrash } from 'react-icons/fa';
+import { useTopicScoreRules } from 'store/hooks/admin';
+import { IAjaxRequest } from 'store/slices';
+import {
+  Button,
+  ButtonVariant,
+  Col,
+  FieldSize,
+  FormikCheckbox,
+  FormikSelect,
+  FormikText,
+  FormikTimeInput,
+  FormPage,
+  IOptionItem,
+  Loader,
+  Row,
+  Show,
+} from 'tno-core';
+
+import { defaultTopicScoreRule } from './constants';
+import { ITopicScoreRuleForm } from './interfaces';
+import * as styled from './styled';
+import { toForm, toModel } from './utils';
+
+/**
+ * Component to display a list of rules.
+ * Provides CRUD methods for rules.
+ * @returns Component
+ */
+export const TopicScoreRuleList: React.FC = () => {
+  const [{ rules }, api] = useTopicScoreRules();
+  const [{ sourceOptions }] = useLookupOptions();
+
+  const [items, setItems] = React.useState<ITopicScoreRuleForm[]>([]);
+  const [isSaving, setIsSaving] = React.useState(false);
+
+  const findAll = React.useCallback(async () => {
+    await api.findAllTopicScoreRules().then((data) => {
+      if (!!data.length) setItems(data.map((r) => toForm(r)));
+      else setItems([defaultTopicScoreRule]);
+    });
+  }, [api]);
+
+  React.useEffect(() => {
+    findAll();
+    // Only load the topic score rules the first time.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const handleSave = async (values: ITopicScoreRuleForm[]) => {
+    if (!isSaving) {
+      try {
+        setIsSaving(true);
+        const results = await api.updateTopicScoreRules(values.map((r) => toModel(r)));
+        setItems(results.map((r) => toForm(r)));
+      } catch {
+        // Errors are handled globally,
```

Comment:
I don't have a way to handle errors when the save fails.  This can happen for a number of reasons.